allennlp/data/dataset_readers/seq2seq.py

```diff
@@ -75,18 +119,31 @@ def _read(self, file_path):
                 if len(line_parts) != 2:
                     raise ConfigurationError("Invalid line format: %s (line number %d)" % (line, line_num + 1))
                 source_sequence, target_sequence = line_parts
-                yield self.text_to_instance(source_sequence, target_sequence)
+                instance = self.text_to_instance(source_sequence, target_sequence)
+                if instance is None:
+                    continue
+                yield instance
 
     @overrides
     def text_to_instance(self, source_string: str, target_string: str = None) -> Instance:  # type: ignore
         # pylint: disable=arguments-differ
         tokenized_source = self._source_tokenizer.tokenize(source_string)
+        if len(tokenized_source) > self._source_max_sequence_length and self._filter_source:
```

Comment:
We've been debating about whether we should put this kind of functionality directly into the `Tokenizer`, as it's a common problem to have.  What do you think?

Some issues:
- Wanting to discard instances completely if they exceed some max length seems like it's functionality that belongs in the `DatasetReader`, not the `Tokenizer` (the tokenizer could return `None` here in those cases, but that seems awkward).  And if you have one `Tokenizer` that operates on both questions and passages in SQuAD, you might want different max lengths for each of those, so you'd want the `DatasetReader` to pass these lengths to `tokenize()`, not have them be parameter that get passed to the `Tokenizer's` constructor.
- On the other hand, `Tokenizers` can add start and end tokens to their strings (which are separate from the `START_SYMBOL` and `END_SYMBOL` here - these are required for decoder functionality, but you might want additional buffer tokens if you're using a CNN to encode things).  So the truncation that you're doing outside of the `Tokenizer` messes with any start and end tokens that the tokenizer may have added.

Any ideas?

Reply:
I'm not as well-versed in the API as you are, so take this with a grain of salt. i do think that it feels more natural for truncation and filtering to be part of the `DatasetReader` --- the `Tokenizer` should only handle tokenizing and little else. Are there currently examples where point #2 shows up, or were you just saying that it'd be a problem for the future if we want to implement a convolutional encoder? Doesn't seem like there's an easy solution either way...

Reply:
Yes, there are, when you use a `CharacterTokenizer`, because these are typically encoded with CNNs (that's where this first came up, with @joelgrus wondering where to put truncation for characters).  With things moving to transformer-style architectures, we might want to have more of these start and end tokens on sentences, also.